### ISSUE TITLE
Triggers block light

### ISSUE THREAD

**Issue by Panjoo** [jdolan/quetoo#281]
`Trigger_` brushes block light, even if I flag the brush as nodraw & trans.
Only trigger_teleporter brushes do not have this problem it seems.

I was working on some one-way teleporters that need to have a trigger_push over the exit so this is annoying again. And since I don't know what's going on it always takes me a good while before I can be 100% sure the problem is not in what I'm doing in the map editor but in quetoo itself.

First time I noticed this was back when I was building the very first version Lavatomb so this still hasn't been fixed. (You can clearly see that the triggers that cover the sliding doors (above Quad) and the 3 sinking steps cast shadows. The pointlights that are placed right above them don't do anything.) 
I remember when I used the clip texture on the triggers instead I started to get HOM's, lol. So at last I just slapped a normal floor texture on the triggers and flagged that as nodraw/trans. But it seems that workaround doesn't work anymore as no matter what I do the triggers don't let light pass.

**Comment by jdolan** [comment]
Noted! @kaadmy do you have any interest in taking this one on? @Panjoo could you attach a minimal test case map that clearly shows the issue (a light above a `CONTENTS_TRIGGER` brush)?

**Comment by kaadmy** [comment]
If this is related to quemap (which it seems to be), I'd prefer to stay away ;)
I thought `trigger_teleport` was a bbox entity without a brush?

**Comment by jdolan** [comment]
`misc_teleport` is a bbox defined by the editor and game statically. `trigger_teleporter`, `trigger_push`, `trigger_hurt` `trigger_exec` use mapper-defined brush models to turn a volume into a specialized trigger. I actually misspoke in my previous reply -- there is no `CONTENTS_TRIGGER`.

In all of these `trigger_*` cases, the brush model itself is explicitly hidden from the user, regardless of any visibility flags. In the game code, when the entity is created, it is set as `SVF_NO_CLIENT` -- so it's literally impossible to see trigger brushes with the current implementation. So there absolutely needs to be some code to `|= CONTENTS_WINDOW` them during BSP compile so that they do not block light.

@Panjoo I'm trying to fix a few critical physics and prediction bugs, but will get to this before long. I apologize that it may take a couple of weeks.

**Comment by Panjoo** [comment]
Alright cool. For now I have just relocated some lights to bypass the shadow problem.

**Comment by jdolan** [comment]
This is a very easy fix. While I agree that the contents flags are confusing, we didn't invent them, and I'm not sure I'd qualify this as a bug.

I just fixed your test map by making the triggers `CONTENTS_MIST`. I removed all other surface and contents flags from them (no `CONTENTS_TRANSLUCENT`, no `SURF_NO_DRAW`, etc). Just good old `CONTENTS_MIST`. They still interact and push the player, without blocking any light. Screenshot attached.

![quetoo031](https://cloud.githubusercontent.com/assets/643118/21490429/7d002366-cbc2-11e6-8df4-8baf4c801265.jpg)

If this works for you, please close out :)

**Comment by Panjoo** [comment]
Okay cool. But it is certainly different from other games I've mapped for. I'm just going by what I've learned to do and I tick these flag combo's on auto-pilot. In this case the trans flag was just a last resort to stop it from blocking light. 

Anyway, in nearly every game you make triggers by using a trigger texture, which will flag it nodraw and treat it like mist but it never blocks light as long as nodraw is involved. I think that's just odd. But if only flagging as mist works in this game then that's fine of course.
Might be an idea to edit the entities.def for this scenario, eg.: "For trigger_ brushes, only use the `CONTENTS_MIST` flag."
